src/providers/storage/assetProvider.ts

```diff
@@ -17,14 +18,15 @@ export class AssetProviderFactory {
         AssetProviderFactory.handlerRegistry[name] = factory;
     }
 
+    public static createFromConnection(connection: IConnection): IAssetProvider {
+        return this.create(connection.providerType, connection.providerOptions);
+    }
+
     public static create(name: string, options?: any): IAssetProvider {
         Guard.emtpy(name);
 
         const handler = AssetProviderFactory.handlerRegistry[name];
         console.log(handler);
-        // if (!handler) {
-        //     throw new Error(`No asset provider has been registered with name '${name}'`);
-        // }
 
         return handler(options);
```

Comment:
I think this would be a good place to instantiate the provicer, then check if it has an initialize method... if so call it before returning the handler.  Only additional though is that it is defined as async (returns promise) which would require any callers to await the creation of the provider.  Let's discuss.

Reply:
Flag in create to initialize or not? But I guess the method signature would still need to be async